source/parser/helper.civet

```diff
@@ -87,6 +88,65 @@ declareHelper := {
       "  }\n"
       "})"
     ], ";\n"]
+  range(rangeRef): void
+    state.prelude.push ["", [
+      preludeVar
+      rangeRef
+      ts [ ": (start: number, end: number) => number[]" ]
+      " "
+      """
+      = (start, end) => {
+        const length = end - start;
+        if (length <= 0) return [];
+        const arr = Array(length);
+        for (let i = 0; i < length; ++i) {
+          arr[i] = i + start;
+        }
+        return arr;
+      }
+      """
+    ], ";\n"]
+  revRange(revRangeRef): void
+    state.prelude.push ["", [
+      preludeVar
+      revRangeRef
+      ts [ ": (start: number, end: number) => number[]" ]
+      " "
+      """
+      = (start, end) => {
+        const length = start - end;
+        if (length <= 0) return [];
+        const arr = Array(length);
+        for (let i = 0; i < length; ++i) {
+          arr[i] = start - i;
+        }
+        return arr;
+      }
+      """
+    ], ";\n"]
+  stringRange(stringRangeRef): void
+    state.prelude.push ["", [
+      preludeVar
+      stringRangeRef
+      ts [ ": (start: number, length: number, step: number) => string[]" ]
+      " "
+      """
+      = (start, length, step) => {
+        if (length <= 0) return [];
+        const arr = Array(length);
+        if (step > 0) {
```

Comment:
Since `step` is treated as a boolean `ascending` we should change it to `ascending: boolean`. Also a test for reversed string range would be good to add as well.